pkg/registry/apis/query/query.go

```diff
@@ -337,3 +369,51 @@ func (b *QueryAPIBuilder) handleExpressions(ctx context.Context, req parsedReque
 	}
 	return qdr, nil
 }
+
+// logRequest short-term hack until k8s have added support for traceIDs in request logs.
+func logRequest(logger log.Logger, startTime time.Time, statusCode int) {
+	duration := time.Since(startTime)
+	logger.Debug("Query request completed", "status", statusCode, "duration", duration.String())
+}
+
+func logRequestError(logger log.Logger, startTime time.Time, err error) {
+	var args []any
+	var gfErr errutil.Error
+	if !errors.As(err, &gfErr) {
+		args = []any{"error", err.Error()}
+	} else {
+		args = []any{
+			"errorReason", gfErr.Reason,
+			"errorMessageID", gfErr.MessageID,
+			"error", gfErr.LogMessage,
+		}
+	}
+
+	duration := time.Since(startTime)
+	args = append(args, "duration", duration.String())
+	logger.Error("Query request completed", args...)
```

Comment:
```suggestion
	logger.Error("Query request failed", args...)
```

Reply:
Decide not to accept this since we do have the level indicating failure and follows our conventions for grafana request logs. This is temporary and hopefully we can get rid of it soon